mne/viz/_3d.py

```diff
@@ -953,14 +954,134 @@ def _limits_to_control_points(clim, stc_data, colormap):
     return ctrl_pts, colormap
 
 
+def _handle_time(time_label, time_unit, times):
+    """Handle time label string and units."""
+    if time_label == 'auto':
+        time_label = 'time=%0.3f ms'
+    if time_unit == 's':
+        times = times
+    elif time_unit == 'ms':
+        times = 1e3 * times
+    else:
+        raise ValueError("time_unit needs to be 's' or 'ms', got %r" %
+                         (time_unit,))
+
+    return time_label, times
+
+
+def _plot_mpl_stc(stc, subject=None, surface='inflated', hemi='lh',
+                  colormap='auto', time_label='auto', smoothing_steps=10,
+                  subjects_dir=None, views='lat', clim='auto', figure=None,
+                  initial_time=0, time_unit='s', background='black',
+                  spacing='oct5'):
+    """Plot source estimate using mpl."""
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D, art3d
+    from matplotlib import cm
+    import nibabel as nib
+    from scipy import sparse, stats
+    from ..source_estimate import _morph_buffer, _get_subject_sphere_tris
+    if hemi not in ['lh', 'rh']:
+        raise ValueError("hemi must be 'lh' or 'rh' when using matplotlib. "
+                         "Got %s." % hemi)
+    kwargs = {'lat': {'elev': 5, 'azim': 0},
+              'med': {'elev': 5, 'azim': 180},
+              'fos': {'elev': 5, 'azim': 90},
+              'cau': {'elev': 5, 'azim': -90},
+              'dor': {'elev': 90, 'azim': 0},
+              'ven': {'elev': -90, 'azim': 0},
+              'fro': {'elev': 5, 'azim': 110},
+              'par': {'elev': 5, 'azim': -110}}
+    if views not in kwargs:
+        raise ValueError("views must be one of ['lat', 'med', 'fos', 'cau', "
+                         "'dor' 'ven', 'fro', 'par']. Got %s." % views)
+    ctrl_pts, colormap = _limits_to_control_points(clim, stc.data, colormap)
+    if colormap == 'auto':
+        colormap = mne_analyze_colormap(clim, format='matplotlib')
+
+    time_label, times = _handle_time(time_label, time_unit, stc.times)
+    if initial_time is None:
+        time_idx = 0
+    else:
+        time_idx = np.argmin(np.abs(times - initial_time))
+    fig = plt.figure() if figure is None else figure
+    ax = Axes3D(fig)
+
+    surf = op.join(subjects_dir, subject, 'surf', '%s.%s' % (hemi, surface))
+    if spacing == 'all':
+        coords, faces = nib.freesurfer.read_geometry(surf)
+        inuse = slice(None)
+    else:
+        stype, sval, ico_surf, src_type_str = _check_spacing(spacing)
+        surf = _create_surf_spacing(surf, hemi, subject, stype, ico_surf,
+                                    subjects_dir)
+        inuse = surf['vertno']
+        faces = surf['use_tris']
+        coords = surf['rr'][inuse]
+        shape = faces.shape
+        faces = stats.rankdata(faces, 'dense').reshape(shape) - 1
```

Comment:
Here I assume that the spacing code doesn't mess with the order of the vertices.